pennylane/transforms/qcut.py

```diff
@@ -210,3 +214,145 @@ def tape_to_graph(tape: QuantumTape) -> MultiDiGraph:
+    as the communication/quotient graph. In the communication graph, each tensor is assigned a node
+    and edges are added between nodes to mark a contraction along an index. The communication graph
+    resulting from the above contraction equation is a complete directed graph.
+
+    In the communication graph provided by :func:`fragment_graph`, edges are composed of
+    :class:`PrepareNode` and :class:`MeasureNode` pairs. To correctly map back to the contraction
+    equation, we must keep track of the order of preparation and measurement indices in each tensor.
+    This order is specified in the ``prepare_nodes`` and ``measure_nodes`` arguments.
+
+    Args:
+        tensors (Sequence): the tensors to be contracted
+        communication_graph (MultiDiGraph): the communication graph determining connectivity between
+            the tensors
+        prepare_nodes (Sequence[Sequence[PrepareNode]]): a sequence of size
+            ``len(communication_graph.nodes)`` that determines the order of preparation indices in
+            each tensor
+        measure_nodes (Sequence[Sequence[MeasureNode]]): a sequence of size
+            ``len(communication_graph.nodes)`` that determines the order of measurement indices in
+            each tensor
+        use_opt_einsum (bool): Determines whether to use the
+            [opt_einsum](https://dgasmith.github.io/opt_einsum/) package. This package is useful for
+            tensor contractions of large networks but must be installed separately using, e.g.,
+            ``pip install opt_einsum``.
+
+    Returns:
+        float or array-like: the result of contracting the tensor network
+
+    **Example**
+
+    We first set up the tensors and their corresponding :class:`~.PrepareNode` and
+    :class`~.MeasureNode` orderings:
+
+    .. code-block:: python
+
+        from pennylane.transforms import qcut
+        import networkx as nx
+        import numpy as np
+
+        tensors = [np.arange(4), np.arange(4, 8)]
+        prep = [[], [qcut.PrepareNode(wires=0)]]
+        meas = [[qcut.MeasureNode(wires=0)], []]
+
+    The communication graph describing edges in the tensor network must also be constructed:
+
+    .. code-block:: python
+
+        graph = nx.MultiDiGraph([(0, 1, {"pair": (meas[0][0], prep[1][0])})])
+
+    The network can then be contracted using:
+
+    >>> qcut.contract_tensors(tensors, graph, prep, meas)
+    38
+    """
+    # pylint: disable=import-outside-toplevel
+    if use_opt_einsum:
+        try:
+            from opt_einsum import contract, get_symbol
+        except ImportError as e:
+            raise ImportError(
+                "The opt_einsum package is required when use_opt_einsum is set to "
+                "True in the contract_tensors function. This package can be "
+                "installed using:\npip install opt_einsum"
+            ) from e
+    else:
+        contract = qml.math.einsum
+        get_symbol = _get_symbol
```

Comment:
It's mainly so that when we use `get_symbol` below, it is the same function name regardless of `use_opt_einsum`.

We could do `from opt_einsum import contract, get_symbol as _get_symbol` and get rid of this line, but I couldn't get that to work for some reason.